### PR TITLE
third-party example widgets

### PR DESCRIPTION
**Pull request by gavinr**

### CODE REVIEW

File: samples/d3/manifest.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "d3",
+  "type": "widget",
+  "version": "1.0.0",
```
Comment (qlqllu) [inline samples/d3/manifest.json:4]: All widgets use the same version make it easy to manage

File: samples/d3/src/runtime/widget.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/** @jsx jsx */
+import { AllWidgetProps, BaseWidget, jsx } from "jimu-core";
+import * as React from "react";
+import * as d3 from "./lib/d3/d3.min.js";
+
+export default class Widget extends BaseWidget<AllWidgetProps<any>, any> {
```
Comment (qlqllu) [inline samples/d3/src/runtime/widget.tsx:6]: I'll update all other widgets to extends from `React.PureComponent`

File: samples/d3/manifest.json (hunk 1 above)
Comment (gavinr) [reply samples/d3/manifest.json]: Thanks, I updated all the `version` properties to be `1.1.0`.

File: samples/d3/src/runtime/widget.tsx (hunk 2 above)
Comment (gavinr) [reply samples/d3/src/runtime/widget.tsx]: Thanks, I updated these to use `React.PureComponent` following your example.

**Pull request merged**